src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,283 @@
+
+
+@register
+class CreateValidSignPackageTask(ValidSignTask):
+    """Create a ValidSign package with signers and documents and send a signing request to the signers.
+
+    **Required process variables**
+
+    * ``documents``: List of strings. List of API URLs where the documents to be signed can be retrieved.
+        The API must comply with the Documenten API 1.0.x (https://vng-realisatie.github.io/gemma-zaken/standaard/documenten/index).
+
+    * ``signers``: JSON list with signers information. For ValidSign, the first name, the last name and the
+        email address of each signer are required. Example ``signers``:
+
+            .. code-block:: json
+
+                [{
+                    "email": "example.signer@example.com",
+                    "firstName": "ExampleFirstName",
+                    "lastName": "ExampleLastName"
+                },
+                {
+                    "email": "another.signer@example.com",
+                    "firstName": "AnotherFirstName",
+                    "lastName": "AnotherLastName"
+                }]
+
+    * ``packageName``: string. Name of the ValidSign package that contains the documents to sign and the signers.
+        This name appears in the notification-email that is sent to the signers.
+
+    **Sets the process variables**
+
+    * ``packageId``: string. ID of the ValidSign package created by the task.
+    """
+
+    _documents_client = None
+
+    def get_documents_client(self, topic_name: str) -> ZGWClient:
+        if self._documents_client is None:
+            default_services = TaskMapping.objects.get(
+                topic_name=topic_name
+            ).defaultservice_set.select_related("service")
+            services_by_alias = {svc.alias: svc.service for svc in default_services}
+
+            alias = "DocumentenAPI"
+            if alias not in services_by_alias:
+                raise RuntimeError(f"Service alias '{alias}' not found.")
+
+            self._documents_client = services_by_alias[alias].build_client()
+        return self._documents_client
+
+    def format_signers(self, signers: List[dict]) -> List[dict]:
+        """Format the signer information into an array of JSON objects as needed by ValidSign."""
+
+        return [{"type": "SIGNER", "signers": [signer]} for signer in signers]
+
+    def _get_documents_from_api(self) -> List[Tuple[str, bytes]]:
+        """Retrieve the documents and their content from the Documenten API."""
+
+        logger.debug("Retrieving documents from Documenten API")
+
+        variables = self.task.get_variables()
+        document_urls = check_variable(variables, "documents")
+        document_client = self.get_documents_client(topic_name="CreateValidSignPackage")
+
+        documents = []
+        for document_url in document_urls:
+            # Retrieving the document
+            document_data = document_client.retrieve(
+                resource="enkelvoudiginformatieobject", url=document_url,
+            )
```

Comment:
This is going to be a rough one - it's not unrealistic that multiple Documents API's are used, for example:
- https://drc.utrechtproeftuin.nl/api/v1/
- https://alfresco.utrechtproeftuin.nl/drc/api/v1/
- https://openzaak.utrechtproeftuin.nl/documenten/api/v1/

Each of those have different credentials. The client should be constructed from `document_url` rather than using a static configured client.

It's a complex situation, especially w/r to #58 in mind - the JWT's have to be fed into the process from the initial applications.

What you _can_ assume is:

- each DRC service is connected to the task mapping (`TaskMapping` and `DefaultService`) with an alias, e.g. `drc1`, `drc2`...
- the process variable `services` contains the JWT's - see how the `zgw` work units do this

Then, you can:

1. for every default service for the task mapping
    1. build the client
2. find the best match within the default services (check if URL starts with `client.api_root`)
3. use that client and use the credentials from the `services` variable

This can be abstracted away properly in zgw-consumers, looking at https://github.com/maykinmedia/zgw-consumers/blob/master/zgw_consumers/query.py#L10 by turning this manager method into a queryset method and filtering the services based on the task default services.